data_model/src/transaction.rs

```diff
@@ -985,14 +1012,45 @@ pub mod error {
             Serialize,
             IntoSchema,
         )]
-        #[display(fmt = "Action not permitted: {reason}")]
-        #[serde(transparent)]
-        #[repr(transparent)]
-        // SAFETY: `NotPermittedFail` has no trap representation in `String`
-        #[ffi_type(unsafe {robust})]
-        pub struct NotPermittedFail {
-            /// The cause of failure.
-            pub reason: String,
+        #[ffi_type(opaque)]
+        #[cfg_attr(feature = "std", derive(thiserror::Error))]
+        pub enum ValidationFailure {
```

Comment:
I see some are named with suffix `Fail` and some with `Failure`

Reply:
Will replace with `ValidationFail`